src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -324,14 +326,14 @@ public static function provideInvalidInputValues()
     /**
      * @dataProvider provideInputValues
      */
-    public function testValidInput($expected, $value)
+    public function testValidInput(?string $expected, null|float|string $value)
```

Comment:
I wouldn't recommend adding types to arguments of test cases. There are two reasons:
 - type juggling might hide an issue (we're not using strict types)
- it's just visual debt since test cases are not an API

Reply:
> type juggling might hide an issue (we're not using strict types)

It would not hide more than before, wouldn't it?

> it's just visual debt since test cases are not an API

For me it also help while reading a test case, but if you want me to revert, I will do it

Reply:
> It would not hide more than before, wouldn't it?

Yes: pass an int, now it's a string, assertSame will see something different. Depending on the situation, this could hide something.

Reply:
> For me it also help while reading a test case, but if you want me to revert, I will do it

Any other opinion on this? My comment was to warn that adding types should be a conscious decision, not just a "best practice I turned my brain off" one ;)

Reply:
I'm all for adding types here. It improves the readability of those tests significantly. I usually add those tests when I gad to reverse-engineer parameter types to understand the test.